### DISCUSSION TITLE
Support for recursive types

### DISCUSSION THREAD

**Discussion by kshefchek** [linkml/linkml#217] [Ideas]
The biolink model entity class has an attribute called category which is essentially a list of biolink model classes (excluding mixins and abstract classes).  However, there's no formalized way to represent this kind of type (not sure if recursive type is the right term).  It would be especially useful for generating defaults, as I've prototyped in the pydantic gen in a hacky way with a _category classvar for non mixin|abstract classes and this post init on Entity:

```
    def __post_init__(self):
        # Initialize default categories if not set
        # by traversing the MRO chain
        if not self.category:
            self.category = list(
                {
                    f'biolink:{super_class._category}'
                    for super_class in inspect.getmro(type(self))
                    if hasattr(super_class, '_category')
                }
            )
```

Is it in scope for linkml to natively provide this type?

**Comment by cmungall** [comment]
See also #95

**Comment by cmungall** [comment]
This is now supported via type_generators
